beamer/tests/agent/unit/test_challenge.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+
+from beamer.chain import process_claims
+from beamer.tests.agent.unit.utils import (
+    ADDRESS1,
+    FILL_ID,
+    TIMESTAMP,
+    make_claim,
+    make_context,
+    make_request,
+)
+from beamer.tests.agent.utils import make_address
+from beamer.typing import FillId, Termination
+
+
+@pytest.mark.parametrize("fill_id", [FILL_ID, FillId(b"cafebabe")])
+@pytest.mark.parametrize("filler", [ADDRESS1, make_address()])
+def test_challenge_as_challenger(filler, fill_id):
+    context, _ = make_context()
+
+    request = make_request()
+    context.requests.add(request.id, request)
+    request.filler = ADDRESS1
+    request.fill_id = FILL_ID
+    claim = make_claim(
+        request=request,
+        claimer=filler,
+        fill_id=fill_id,
+        termination=Termination(TIMESTAMP + 1),
+    )
+    context.claims.add(claim.id, claim)
+
+    honest_claim = claim.valid_claim_for_request(request)
+    assert not claim.transaction_pending
+    process_claims(context)
+    assert honest_claim != claim.transaction_pending
+
+
+def test_challenge_as_claimer():
+    context, _ = make_context()
+
+    request = make_request()
+    context.requests.add(request.id, request)
+    request.filler = make_address()
+    claim = make_claim(
+        request=request,
+        claimer=context.address,
+        challenger=make_address(),
```

Comment:
you don't strictly need this, it will be set automatically.

Reply:
for claimer you are right, but I think for challenger it's not the case (can be None)

Reply:
yeah just came across that .. I think, I will change that in order to be able to create an unchallenged claim.